src/main/java/com/evanlennick/retry4j/CallExecutor.java

```diff
@@ -191,17 +191,24 @@ private boolean shouldThrowException(Exception e) {
                 return false;
             }
 
+            Class<?> thrownExceptionClass = e.getClass();
+            if(this.config.shouldMatchAgainstExceptionCause()){
+                if(e.getCause() != null) {
+                    thrownExceptionClass = e.getCause().getClass();
```

Comment:
I don't really like how narrow this is... if I'm understanding it correctly, it's going to cause retries on the next exception down the cause chain but won't be affected by the top level or any causes further down the exception chain. I have thought about this scenario before and there is an issue here:

https://github.com/elennick/retry4j/issues/59

What I would actually prefer as a solution is if there was a more general config toggle called something like `retryOnCausedBy()` instead of `shouldMatchAgainstExceptionCause()` and it caused retries to trigger when that exception was anywhere in the caused-by chain. Let me know what you think about modifying this to do that.

Reply:
- Yes I do like to expand the search to match `cause` at any level as I also did suggest initially. I did make this pull request based on issue #59. 
   - I believe by this approach we also mean that if there is no cause for a top-level exception, then `UnexpectedException` would be thrown.
- Would you prefer to name the config param to be `retryOnCausedBy` instead of `matchAgainstExceptionCause`?